### PR TITLE
[MAINT] Mark bonding experimental

### PR DESCRIPTION
**Pull request by ethouris**

### CODE REVIEW

File: srtcore/core.cpp [hunk 1]
```diff
@@ -174,8 +174,8 @@ void CUDT::construct()
     // Initilize mutex and condition variables
     initSynch();
 
-    // Default: 
-    m_cbPacketArrival.set(this, &CUDT::defaultPacketArrival);
+    // XXX: Unblock, when the callback is implemented
+    // m_cbPacketArrival.set(this, &CUDT::defaultPacketArrival);
```
Comment (maxsharabayko) [inline srtcore/core.cpp:178]: Why this change is here?

File: srtcore/core.cpp [hunk 2]
```diff
@@ -10877,6 +10939,8 @@ void CUDT::removeEPollID(const int eid)
     leaveCS(s_UDTUnited.m_EPoll.m_EPollLock);
 }
 
+#if ENABLE_EXPERIMENTAL_BONDING
+#endif
```
Comment (maxsharabayko) [inline srtcore/core.cpp:10943]: ?

File: srtcore/srt.h [hunk 3]
```diff
@@ -772,6 +757,26 @@ SRT_API       SRTSOCKET srt_create_socket(void);
 
 // Group management
 
+// Stubs when off
+
+typedef struct SRT_SocketGroupData_ SRT_SOCKGROUPDATA;
```
Comment (maxsharabayko) [inline srtcore/srt.h:762]: Maybe just `typedef void SRT_SOCKGROUPDATA` if `ENABLE_EXPERIMENTAL_BONDING` is not defined. Then the structure `SRT_SocketGroupData_` can still be
```c++
typedef struct SRT_SocketGroupData_ {
   // ..
} SRT_SOCKGROUPDATA;
```

File: srtcore/core.cpp (hunk 1 above)
Comment (ethouris) [reply srtcore/core.cpp:178]: Because it's not implemented. This is a stub that should be filled with an implementation that processes the packet for the sake of group receiver buffer, if provided. Also the call to this callback isn't currently properly defined.

File: srtcore/srt.h (hunk 3 above)
Comment (ethouris) [reply srtcore/srt.h:762]: What's exactly the advantage, beside that the `SRT_SOCKGROUPDATA` will not be a unique type name?

Comment (maxsharabayko) [reply srtcore/srt.h:762]: Now some undefined `struct SRT_SocketGroupData_` is used if bonding is disabled.

Comment (ethouris) [reply srtcore/srt.h:762]: Just like the `FILE` symbol in the standard C library.

**Comment by maxsharabayko** [comment]
Build failures in AppVeyor
```shell
(ClCompile target) -> 
  C:\projects\srt\srtcore\core.cpp(8815,1): error C2121: '#': invalid character: possibly the result of a macro expansion [C:\projects\srt\_build\srt_virtual.vcxproj]
  C:\projects\srt\srtcore\core.cpp(8809,1): error C2143: syntax error: missing ';' before 'if' [C:\projects\srt\_build\srt_virtual.vcxproj]
  C:\projects\srt\srtcore\core.cpp(8815,1): error C2061: syntax error: identifier 'ENABLE_EXPERIMENTAL_BONDING' [C:\projects\srt\_build\srt_virtual.vcxproj]
```

**Review by maxsharabayko — APPROVED** [review]
